src/switchroot/ostree-prepare-root.c

```diff
@@ -325,6 +325,25 @@ main (int argc, char *argv[])
         err (EXIT_FAILURE, "failed to bind mount (class:readonly) /usr");
     }
 
+  /* Prepare /sysroot.
```

Comment:
One thing that may help here is for us to use the new mount API. I am pretty sure we can reasonably hard require it now.

With that, we no loner need `sysroot.tmp` and these absolute paths - it becomes easier to reason about the mount setup. Also we can construct the entire mount as a transactional unit (detached fd) that we then attach in place.